### PR TITLE
Switched from SCMTrigger.SCMTriggerCause to BranchIndexingCause

### PR DESCRIPTION
**Pull request by jglick**
`SCMTriggerCause` is not really appropriate for multibranch projects. For example, it tries to save a polling log, which in fact has bogus text. And the hyperlink is not useful.

@reviewbybees

### CODE REVIEW

**Comment by recena** [comment]
I like the idea.

File: src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:f="/lib/form">
+    <j:set var="url" value="${it.indexingUrl}"/>
+    <j:choose>
+        <j:when test="${url != null}">
+            <a href="${rootURL}/${url}console">Branch indexing</a>
+        </j:when>
+        <j:otherwise>
+            Branch indexing
```
Comment (recena) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: Does it make sense to show in the left-menu a entry without link?

Comment (jglick) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: This is not in the sidebar.

Comment (recena) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: Where is used this _part of_ view? or How is used?

Comment (jglick) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: Read the Javadoc for `Cause`.

**Comment by kwhetstone** [comment]
:bee: Makes sense; do we need some tests for this?

**Comment by jglick** [comment]
Currently it is not used at any API level, just a minor UI feature, but I am looking into using it from a variant of #1, which would get test coverage.

File: src/main/java/jenkins/branch/MultiBranchProject.java [hunk 2]
```diff
@@ -363,15 +361,15 @@ public void observe(@NonNull SCMHead head, @NonNull SCMRevision revision) {
...
+        if (ParameterizedJobMixIn.scheduleBuild2(item, 0, new CauseAction(new BranchIndexingCause())) != null) {
             listener.getLogger().println("Scheduled build for branch: " + name);
             try {
                 factory.setRevisionHash(item, revision);
             } catch (IOException e) {
                 e.printStackTrace(listener.error("Could not update last revision hash"));
             }
         } else {
-            listener.getLogger().println("Failed to schedule build for branch: " + name);
+            listener.getLogger().println("Did not schedule build for branch: " + name);
```
Comment (jtnord) [inline src/main/java/jenkins/branch/MultiBranchProject.java]: why the change?
to me "did not" means there was nothing to do whereas "failed" is I tried but couldn't.
"Failed" seems more appropriate here (Jenkins was shutting down perhaps?)

Comment (jglick) [inline src/main/java/jenkins/branch/MultiBranchProject.java]: In the context of a possible alternative to #1, this message is more accurate.

A null return value could mean several things (unfortunately `ScheduleResult.Refused` does not differentiate them); the most innocuous would be a `QueueDecisionHandler` that returns `false` from `shouldSchedule` (again, unfortunately, it is not given a `TaskListener` to explain its decision, or offered to return a `CauseOfBlockage` or similar).

Comment (jtnord) [inline src/main/java/jenkins/branch/MultiBranchProject.java]: thanks.

**Comment by jtnord** [comment]
:bee:

File: src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly (hunk 1 above)
Comment (recena) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: @jglick From the Javadoc in `Cause`...

>  Cause object base class.  This class hierarchy is used to keep track of why a given build was started. This object encapsulates the UI rendering of the cause, as well as providing more useful information in respective subypes.

I don't still know who Jelly file includes to this one. Could you help me?

Comment (jglick) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: > From the Javadoc in `Cause`...

Read a little longer! :-)

> <h2>Views</h2>
> <dl>
> <dt>description.jelly
> <dd>Renders the cause to HTML. By default, it puts the short description.
> </dl>

So…

> I don't still know who Jelly file includes to this one.

`CauseAction/summary.jelly`

Comment (recena) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: @jglick If I'm asking is because it was not clear to me ;(

Comment (recena) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: Ah ok, this jelly file replaces to this one https://github.com/jenkinsci/jenkins/blob/master/core/src/main/resources/hudson/model/Cause/description.jelly.

Thanks.

**Comment by recena** [comment]
Belated 🐝

File: src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly (hunk 1 above)
Comment (jglick) [inline src/main/resources/jenkins/branch/BranchIndexingCause/description.jelly]: Yes, the default prints plain text, but in this we would like a hyperlink.

File: src/main/java/jenkins/branch/MultiBranchProject.java (hunk 2 above)
Comment (jglick) [inline src/main/java/jenkins/branch/MultiBranchProject.java]: [JENKINS-35403](https://issues.jenkins-ci.org/browse/JENKINS-35403)
